tensorflow/core/kernels/conv_grad_filter_ops.cc

```diff
@@ -981,9 +987,61 @@ void LaunchConv2DBackpropFilterOp<Eigen::GpuDevice, T>::operator()(
   cudnn_use_autotune = true;
 #endif
   AlgorithmConfig algorithm_config;
-  if (cudnn_use_autotune && !AutoTuneConvBwdFilter::GetInstance()->Find(
-                                conv_parameters, &algorithm_config)) {
+  bool do_autotune;
+#if GOOGLE_CUDA
```

Comment:
Is this #if necessary? In the case of legacy cuDNN or MIOpen, CudnnUseFrontend always returns false, right?

In general we prefer runtime if statements than #if, because we have more unified build results without #if.

Can you also check all other #if and change accordingly?

Reply:
Done.